node/coinstacks/solana/api/src/controller.ts

```diff
@@ -73,9 +79,67 @@ export class Solana implements BaseAPI {
   @Response<ValidationError>(422, 'Validation Error')
   @Response<InternalServerError>(500, 'Internal Server Error')
   @Get('account/{pubkey}/txs')
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  async getTxHistory(@Path() pubkey: string, @Query() _cursor?: string, @Query() _pageSize = 10): Promise<TxHistory> {
-    return { pubkey } as TxHistory
+  async getTxHistory(
+    @Path() pubkey: string,
+    @Query() lastSignature?: string,
+    @Query() pageSize = 10
+  ): Promise<TxHistory> {
+    const urlParams = new URLSearchParams()
+    urlParams.append('api-key', RPC_API_KEY ?? '')
+    urlParams.append('limit', pageSize.toString())
+    if (lastSignature) {
+      urlParams.append('before', lastSignature)
+    }
+
+    try {
+      const { data } = await axiosNoRetry.get<RawTx[]>(
+        `${INDEXER_URL}/v0/addresses/${pubkey}/transactions/?${urlParams.toString()}`
+      )
+
+      const txs = data.map((tx) => {
+        return {
+          txid: tx.signature,
+          blockHeight: tx.slot,
+          ...tx,
+        }
+      })
+
+      return { pubkey, txs: txs }
+    } catch (err) {
+      throw handleError(err)
+    }
+  }
+
+  /**
+   * Get transaction by txid
+   *
+   * @param {string} txid transaction id
+   *
+   * @returns {Promise<Tx>} transaction history
+   */
+  @Response<BadRequestError>(400, 'Bad Request')
+  @Response<ValidationError>(422, 'Validation Error')
+  @Response<InternalServerError>(500, 'Internal Server Error')
+  @Get('tx/{txid}')
+  async getTxById(@Path() txid: string): Promise<Tx> {
+    const urlParams = new URLSearchParams()
+    urlParams.append('api-key', RPC_API_KEY ?? '')
+
+    try {
+      const { data } = await axiosNoRetry.post<RawTx>(`${INDEXER_URL}/v0/transactions/?${urlParams.toString()}`, {
+        transactions: [txid],
+      })
+
+      const tx = {
+        txid: data.signature,
+        blockHeight: data.slot,
```

Comment:
This seems wrong as slot !== blockHeight but I didn't find a way to get the blockHeight for a specific TX

Also blockHash is not a part of the helius api but a part of `getTransaction` rpc call, if we need it we might need to call it separately

Reply:
Seen in PM, data.slot seems acceptable enough for now